ksql-rest-app/src/main/java/io/confluent/ksql/rest/server/validation/RequestValidator.java

```diff
@@ -108,6 +109,11 @@ public int validate(
       final ConfiguredStatement<?> configured = ConfiguredStatement.of(
           prepared, scopedPropertyOverrides, ksqlConfig);
 
+      final Class<? extends Statement> statementClass = prepared.getStatement().getClass();
```

Comment:
will https://github.com/confluentinc/ksql/issues/3742 solve the need for this PR? after looking at this patch, it seems very out of place to have this in the validator!

Reply:
we still have to discuss that and I expect we will hash it out over the next  week or so. So may not have time to land before the release cut. In this scenario, its better to have some fix landed first and then iterate? We should make a note and undo these bandaids +1 on that.

Reply:
Yes,  #3742 solve the need for this PR. I added the check in the validator because PR https://github.com/confluentinc/ksql/pull/3711 also requires this check and I didn't want to do it twice.